cmd/kubeadm/app/apis/kubeadm/env.go

```diff
@@ -25,7 +25,7 @@ import (
 
 // TODO(phase2) use componentconfig
 // we need some params for testing etc, let's keep these hidden for now
-func GetEnvParams() map[string]string {
+func GetEnvParams(env map[string]string) map[string]string {
```

Comment:
Passing a map in here is a bit awkward.  That means we have to pass a environment map down to anything that might use it.

At the very least, I'd do the following:
* Define a real type for the environment.  Either keep it a map `type EnvParams map[string]string]` or create a real struct in types.go.
* Have GetEnvParams just return one of those.  No need to pass anything in.

Then you can pass that more strongly typed and self documenting thing down.  Or we can keep it global but a bit cleaner with something like:

```
type EnvParams struct {
  KubernetesDir string
  ...
}

func GetEnvParams() *EnvParams {
...
}
```

You can then doctor it how you like and pass it down.  Or we can keep this as "global" but have it as an easy thing to swap in and out/duplicate.

```
var GlobalEnvParams = GetEnvParams()
```

And then in the test:
```
oldEnv := GlobalEnvParams
GlobalEnvParams = GetEnvParams()
GlobalEnvParams.foo = bar
defer func () { GlobalEnvParams = oldEnv }
```